include/ck_tile/core/tensor/buffer_view.hpp

```diff
@@ -1360,4 +1272,71 @@ make_buffer_view(T* p, BufferSizeType buffer_size, X invalid_element_value)
         p, buffer_size, invalid_element_value};
 }
 
+// Free print functions for all buffer_view variants
+template <typename T, typename BufferSizeType, bool InvalidElementUseNumericalZeroValue>
+CK_TILE_HOST_DEVICE static void
+print(const buffer_view<address_space_enum::generic,
+                        T,
+                        BufferSizeType,
+                        InvalidElementUseNumericalZeroValue,
+                        amd_buffer_coherence_enum::coherence_default>& bv)
+{
+    printf("buffer_view{AddressSpace: generic, p_data_: %p, buffer_size_: ",
```

Comment:
To avoid having to specialize on each address_space_enum value, we can create a helper function like the following:

`template<address_space_enum eVal>
 std::string enum_to_string()
{
   if constexpr (eVal == address_space_enum::generic)
   {
     return "generic";
    }
    else if constexpr ...
}
`
That would help consolidate implementations and reduce repetition